### PR TITLE
added click options to perform tap, doubletap, longpress, tap with of…

### PR DESCRIPTION
**Pull request by amuthansakthivel**
Added AppiumClickOptions to perform tap, doubleTap, longPress and TapWithCoordinates
Added tests to verify the commands
Changed return type to SelenideElement to accommodate Command<SelenideElement>

### CODE REVIEW

**Comment by asolntsev** [comment]
@amuthansakthivel I made method `Util.firstOf()` public, published `Selenide 6.10.0-SNAPSHOT` to maven repo.

**Comment by amuthansakthivel** [comment]
@asolntsev - After pulling the latest snapshot version - i get this error 

  public Void execute(SelenideElement proxy, WebElementSource locator, @Nullable Object[] args) {
              ^
  return type Void is not compatible with SelenideElement
  where T is a type-variable:
    T extends Object declared in interface Command

**Comment by asolntsev** [comment]
> i get this compile time error

You just need to change `public Void execute(` to `public SelenideElement execute(`.
And also change `return null` to `return proxy`.

**Comment by amuthansakthivel** [comment]
> > i get this compile time error
> 
> You just need to change `public Void execute(` to `public SelenideElement execute(`. And also change `return null` to `return proxy`.

Thanks @asolntsev - I know I have to make these changes but wanted to confirm whether we want to do this. Pushed the code now.

File: src/main/java/com/codeborne/selenide/appium/AppiumClickOptions.java [hunk 1]
```diff
@@ -0,0 +1,50 @@
+package com.codeborne.selenide.appium;
+
+import com.codeborne.selenide.ClickMethod;
+import com.codeborne.selenide.ClickOptions;
+import com.codeborne.selenide.appium.commands.AppiumClickMethod;
+
+public class AppiumClickOptions extends ClickOptions {
+
+  private final AppiumClickMethod appiumClickMethod;
+  private final int xOffset;
```
Comment (asolntsev) [inline src/main/java/com/codeborne/selenide/appium/AppiumClickOptions.java:10]: @amuthansakthivel I guess you don't need `offsetX` and `offsetY` in this class?
Parent class `ClickOptions` already has these fields.

**Review by asolntsev — CHANGES_REQUESTED** [review]

File: src/main/java/com/codeborne/selenide/appium/AppiumClickOptions.java (hunk 1 above)
Comment (amuthansakthivel) [reply src/main/java/com/codeborne/selenide/appium/AppiumClickOptions.java:10]: @asolntsev Those are private. It has to be made as protected in order for me to use them.

Comment (asolntsev) [reply src/main/java/com/codeborne/selenide/appium/AppiumClickOptions.java:10]: @amuthansakthivel No, the fields are private, BUT you can still
1. pass these values to constructor (it's protected), and
2. get these values using getters `offsetX()`, `offsetY()` (they are also protected).

Comment (amuthansakthivel) [reply src/main/java/com/codeborne/selenide/appium/AppiumClickOptions.java]: oops my bad. How did I missed it. Fixed it now

**Review by asolntsev — APPROVED** [review]

**Pull request merged**